src/platform_impl/linux/wayland/seat/pointer/handlers.rs

```diff
@@ -153,11 +158,10 @@ pub(super) fn handle_pointer(
             };
 
             let button = match button {
-                0x110 => MouseButton::Left,
-                0x111 => MouseButton::Right,
-                0x112 => MouseButton::Middle,
-                // TODO - figure out the translation.
-                _ => return,
+                BTN_LEFT => MouseButton::Left,
+                BTN_RIGHT => MouseButton::Right,
+                BTN_MIDDLE => MouseButton::Middle,
+                button => MouseButton::Other((button - BTN_LEFT) as u8),
```

Comment:
Can there be any buttons that are smaller `0x110`, or why does it not start at `0`? If so, that would cause troubles, right?

Reply:
I think it could happen but I'm not sure how to handle that without changing winit's `Other` type, since button is u16, not u8. And I have no clue if kernel uses certain things for mouse buttons and not e.g. numpad.

Reply:
But if it matters, I took this logic from glfw, and they subtract left mouse button here if it matters. In theory I can convert everything to u16 and it'll 100% work.

Reply:
I mean at the very least it shouldn't cause an overflow like it's currently doing. Reporting an incorrect button seems much worse than reporting no buttons at all and I wouldn't want to crash in debug mode.

Unfortunately the button values seem kinda odd to me, usually I would have thought a right shift or bitwise ops could have removed all pointless bits, but this `0x110` value seems a bit arbitrary. Unless the button value is always the last 4 bits of `button`.

Reply:
Just look at the include file I've mentioned on the issue, they are really arbitrary.

Reply:
That definitely looks like those buttons are a thing though. So we either have to filter them or remap them I suppose. But we shouldn't rely on overflow.

Reply:
I'll just bump things to u16.